ShellPkg/Library/UefiShellLevel1CommandsLib/If.c

```diff
@@ -895,6 +899,10 @@ ShellCommandRunIf (
   // Make sure that an End exists.
   //
   CurrentScriptFile = ShellCommandGetCurrentScriptFile ();
+  if (CurrentScriptFile == NULL) {
+    return (SHELL_DEVICE_ERROR);

```

Comment:
Switched to `INVALID_PARAMETER`. I think `DEVICE_ERROR` was used because the next return in this function returns `DEVICE_ERROR`. Will follow same behavior in any other calls to `ShellCommandGetCurrentScriptFile()` updated in this PR.